recipes/libffi/all/conanfile.py

```diff
@@ -126,8 +128,21 @@ def _configure_autotools(self):
         self._autotools.configure(args=config_args, configure_dir=self._source_subfolder, build=build, host=host)
         return self._autotools
 
+    @property
+    def _user_info_build(self):
+        # If using the experimental feature with different context for host and
+        # build, the 'user_info' attributes of the 'build_requires' packages
+        # will be located into the 'user_info_build' object. In other cases they
+        # will be located into the 'deps_user_info' object.
+        return getattr(self, "user_info_build", None) or self.deps_user_info
```

Comment:
force newer Conan version only, CCI expects the latest Conan client version. Use `required_conan_version = ">=1.29"` instead.

Reply:
I stole this from https://github.com/conan-io/conan-center-index/blob/eabd1ebb8408f1a328d6779362292cb2f98bcffc/recipes/libelf/all/conanfile.py#L90-L97

Reply:
Ok addressed

Reply:
@uilianries This definitely worked on my ubuntu machine and my M1 mac. But CI is unhappy with this change.

Reply:
@jmarrec the error on CI reminds me a bug with evil insertions: https://github.com/conan-io/conan/pull/8519, I don't think it's because of your changes.
/cc @jgsogo